### DISCUSSION TITLE
Program stalling at "Predicting spectra and IMs"?

### DISCUSSION THREAD

**Discussion by adi772** [vdemichev/DiaNN#454] [Q&A]
Hi Dr Demichev,
I am using DIA-NN for the first time. After hitting 'run' after selecting my input files and the FASTA files, the program runs but then seems to stall. I am not sure how long it should take but I have waited several hours (on three separate occasions now), and the log never proceeds beyond "Predicting spectra and IMs". I have also tried with 'Ultra-fast' setting.

I'm not sure if the program is working in the background. The GUI says "Running..." and the program is using ~4200MB of RAM. Log pasted below (full log attached as .txt file).

Thank you in advance for any help!

8 files will be processed
[0:00] Loading FASTA C:\Users\adi772\Desktop\uniprot-compressed_true_download_true_format_fasta_query__28proteome-2022.07.28-04.43.19.28.fasta\uniprot-compressed_true_download_true_format_fasta_query__28proteome-2022.07.28-04.43.19.28.fasta
[0:15] Processing FASTA
[0:34] Assembling elution groups
[0:54] 9748291 precursors generated
[0:54] Gene names missing for some isoforms
[0:54] Library contains 79611 proteins, and 20506 genes
[0:55] Encoding peptides for spectra and RTs prediction
[1:15] Predicting spectra and IM

[DIANN error log 28-7-22 adi772.txt](https://github.com/vdemichev/DiaNN/files/9206932/DIANN.error.log.28-7-22.adi772.txt)
![adi772 DIANN stalling screenshot 28-7-22](https://user-images.githubusercontent.com/110159670/181455616-287dcae9-fe76-40a4-9854-134ac4b8f86d.PNG)
s

**Comment by vdemichev** [comment]
How does the output look like with Log level set to 3?

**Reply by vdemichev** [reply]
How much RAM does the PC have?

**Reply by adi772** [reply]
PC has 16GB RAM. I have noticed sometimes DIA-NN consumes 9-10GB, but not with 'ultra-fast' setting on.

**Reply by adi772** [reply]
I let the program run overnight. It seems it takes 644 min to generate the library. Is this longer than expected?

Inspecting the log, it seems that none of my data files could be loaded. In each of the 8 files, the following error appears:
"No MS2 spectra: aborting
ERROR: cannot load the file, skipping".

I have attached the log as .txt file.
[report.log.txt](https://github.com/vdemichev/DiaNN/files/9215310/report.log.txt)

**Reply by vdemichev** [reply]
Yes, this is too long, maybe not enough RAM and it's swapping to disk. 
About .raw files, please check that Thermo MSFileReader is installed, specifically using the link provided here https://github.com/vdemichev/DiaNN#raw-data-formats

**Reply by Aaron-Amunix** [reply]
Maybe try converting the raw files (via MSConvert) to mzML, I do this with my Thermo raw files and it works quite well.

**Reply by adi772** [reply]
Thank you both for your replies. It is greatly appreciated. I will try both of these approaches.
